Display fake data for nonLDAP users

TrangEmily opened the pull request:
This PR is for issue #66 
Hi @armenzg, I "reverted" two diffs you showed in the issue's description. I also changed a little bit in function `findReportees` of `getAllFakeReportees.js`, because the user's email is not in fake data. This is the change:

```javascript
 const allEmails = Object.keys(completeOrg);
 const fakeEmail = (email in completeOrg) ? email : allEmails[allEmails.length - 1];

 allReportees[email] = completeOrg[fakeEmail];
 const { reportees } = completeOrg[fakeEmail];
```

However, the result looks different from the old version, so maybe I did something wrong because I don't understand the context. Please help me to check. Thank you!

This is the result of this PR

<img width="525" alt="Screen Shot 2019-04-15 at 4 31 49 PM" src="https://user-images.githubusercontent.com/42683686/56172068-adfbad80-5f9c-11e9-8179-42feeb7af2ce.png">
<img width="555" alt="Screen Shot 2019-04-15 at 4 31 43 PM" src="https://user-images.githubusercontent.com/42683686/56172069-adfbad80-5f9c-11e9-9b07-f4ec90270593.png">
<img width="532" alt="Screen Shot 2019-04-15 at 4 31 35 PM" src="https://user-images.githubusercontent.com/42683686/56172070-adfbad80-5f9c-11e9-8771-cceaa364fcb7.png">

And this is the result of the master branch before you merged the PR about "SSO aunthentication and fetch Taskcluster" 
<img width="538" alt="Screen Shot 2019-04-15 at 3 42 01 PM" src="https://user-images.githubusercontent.com/42683686/56172198-2febd680-5f9d-11e9-895f-f22c76204360.png">
<img width="509" alt="Screen Shot 2019-04-15 at 3 41 54 PM" src="https://user-images.githubusercontent.com/42683686/56172199-2febd680-5f9d-11e9-84fa-ef055cb1d86c.png">
<img width="498" alt="Screen Shot 2019-04-15 at 3 41 47 PM" src="https://user-images.githubusercontent.com/42683686/56172201-2febd680-5f9d-11e9-84ee-e58269695fac.png">
<img width="1146" alt="Screen Shot 2019-04-15 at 4 37 58 PM" src="https://user-images.githubusercontent.com/42683686/56172205-32e6c700-5f9d-11e9-859c-b9d57f5d33b2.png">

armenzg commented on src/views/Main/index.jsx:
```diff
@@ -73,7 +74,7 @@ class MainContainer extends Component {
     async getReportees(userSession, ldapEmail) {
       const secretsClient = userSession.getTaskClusterSecretsClient();
-      const partialOrg = await getAllReportees(secretsClient, ldapEmail);
+      const partialOrg = (userSession.oidcProvider === 'mozilla-auth0') ? await getAllFakeReportees(ldapEmail) : await getAllReportees(secretsClient, ldapEmail);
```
Call `getAllReportees` with `userSession` and make this changes within `getAllReportees`:
https://github.com/mozilla/bugzilla-dashboard/blob/master/src/utils/getAllReportees.js#L52

If you do this you will not need the files `getAllFakeReportees` and `getOrgChart`.

TrangEmily commented:
Thank @armenzg I updated, however, the number still is different from the old version. Is it ok? Thanks.
<img width="505" alt="Screen Shot 2019-04-16 at 8 53 05 AM" src="https://user-images.githubusercontent.com/42683686/56224923-40479400-6025-11e9-99f2-5012713d138f.png">
<img width="528" alt="Screen Shot 2019-04-16 at 8 52 52 AM" src="https://user-images.githubusercontent.com/42683686/56224924-40479400-6025-11e9-8d79-8a100cb17c40.png">

armenzg commented on src/utils/getAllReportees.js:
```diff
@@ -49,8 +56,16 @@ const findReportees = (completeOrg, email) => {
+  } else {
+    const secretsClient = userSession.getTaskClusterSecretsClient();
+    completeOrg = await getOrgChart(secretsClient);
+  }
```
Do this instead:

```javascript
  let people;
  if (userSession.oidcProvider === 'mozilla-auth0') {
    // if non-LDAP user, get fake data
    people = await (await fetch('people.json')).json();
    
  } else {
    const secretsClient = userSession.getTaskClusterSecretsClient();
    const { secret } = await await secretsClient.get(config.taskclusterSecrets.orgData);
    people = secret.employees;
  }
  completeOrg = await buildOrgChartData(people);
```

You can then get rid of the `getOrgChart` function.

armenzg commented:
Please fix the message in the console:
> Warning: Failed prop type: The prop `ldapEmail` is marked as required in `Reportees`, but its value is `undefined`.

Once we have an ldapEmail set in the state we can then create a fake team for it under src/config.js

armenzg commented:
I've verified that immedtiately after signing in I get this message [1]
I'm using GitHub to sign in.
I see the issue both on Firefox and Chrome.

[1]
```
Warning: Failed prop type: The prop `ldapEmail` is marked as required in `Reportees`, but its value is `undefined`.
    in Reportees (created by WithStyles(Reportees))
    in WithStyles(Reportees) (created by Route)
    in Route (at PropsRoute/index.jsx:5)
    in PropsRoute (at Main/index.jsx:258)
    in Switch (at Main/index.jsx:256)
    in Suspense (at Main/index.jsx:255)
    in div (at Main/index.jsx:245)
    in div (at Main/index.jsx:240)
    in MainContainer (created by WithStyles(MainContainer))
    in WithStyles(MainContainer) (created by Route)
    in Route (at PropsRoute/index.jsx:5)
    in PropsRoute (at App/index.jsx:98)
    in Switch (at App/index.jsx:89)
    in div (created by App)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App/index.jsx:84)
    in App (created by WithStyles(App))
    in WithStyles(App) (created by HotExportedWithStyles(App))
    in AppContainer (created by HotExportedWithStyles(App))
    in HotExportedWithStyles(App) (at src/index.jsx:5)
```

haint504 commented:
I updated to get the fake data for Teams tab by using TEAM_CONFIG. Please help me to guide which I should do next. Thank you.
<img width="1288" alt="Screen Shot 2019-04-27 at 10 39 13 PM" src="https://user-images.githubusercontent.com/45067856/56859356-44858280-693e-11e9-86ab-77ef42a2c14e.png">

TrangEmily commented:
I updated to get the fake data for Teams tab by using TEAM_CONFIG. Please help me to guide which I should do next. Thank you.

<img width="1288" alt="Screen Shot 2019-04-27 at 10 39 13 PM" src="https://user-images.githubusercontent.com/42683686/56859477-713a9980-6940-11e9-918c-0b9f866330d0.png">
P/s I deleted the previous message (I used another github account to test, and made a mistake commenting using that account).

sylvestre commented:
Looks great. Armen is off for a few weeks. Let me check who can review that :)

sylvestre commented:
@helfi92 could you please review this please ? :) Thanks

helfi92 commented on src/views/Main/index.jsx:
```diff
@@ -170,29 +169,36 @@ class MainContainer extends Component {
-        }
-      });
+    async teamsData(userSession, partialOrg) {
+      console.log('teams data', userSession, partialOrg);
```
Remove `console.log`.

helfi92 commented on src/views/Main/index.jsx:
```diff
@@ -170,29 +169,36 @@ class MainContainer extends Component {
+        teamComponents = TEAMS_CONFIG;
+      } else {
+        Object.entries(TEAMS_CONFIG).map(async ([teamKey, teamInfo]) => {
+          console.log('team config', teamKey, teamInfo);
```
Remove `console.log`.

TrangEmily commented:
Thank you I removed the "console.log".

sylvestre commented:
Oups, this broke the production.

Here is what I see when I look at the prod:
![image](https://user-images.githubusercontent.com/733326/57159817-42606f80-6de7-11e9-9427-79c15c5bbcb4.png)

I will revert the change, sorry :(

sylvestre commented:
@TrangEmily do you need help ? :)

TrangEmily commented:
Hi @sylvestre, sorry I did not answer sooner. I am out of town and have limited Internet connection. Can I get back next week? Thank you.

sylvestre commented:
sure, thanks!
